### PR TITLE
Added support for configurable reserved usernames

### PR DESCRIPTION
**Pull request by ngerakines**
This PR includes changes to the account model to support the `reserved_usernames` validation check for local accounts. With it, a list of usernames can be added to settings.yml as "reserved" for the purpose of preventing people from registering accounts with those usernames.

### CODE REVIEW

File: config/settings.yml
Comment (ngerakines) [inline config/settings.yml]: I added "admin" and "support" as default entries because those are two account names that I wouldn't want people to steal.

Comment (wxcafe) [inline config/settings.yml]: Please add the other ones that were discussed on #1355

Comment (ngerakines) [inline config/settings.yml]: Done.

**Review by Fastidious — COMMENTED** [review]
How those get unreserved?

**Comment by ngerakines** [comment]
Removing the configuration entry will allow the account to be created.

**Comment by wxcafe** [comment]
It would be nice to be able to alias these accounts to another one, but this is a much bigger feature. I'll merge this when we know why the CI doesn't want to build (it's apparently not related to this tho) and I'll change the issue to reflect the change in scope

**Comment by ngerakines** [comment]
Yeah, I saw the comment about aliasing, but decided to hold off on that one. A temporary solution for web visitors could be to have nginx/httpd rewrite rule samples.

Locally `rake` and spec file execution all looks good. I'm not sure what is going on with travis.

Thanks for taking a look and I'm glad it is useful.

**Comment by ngerakines** [comment]
If there is anything that I can do to help this move along, please let me know. Looks like there is quite the backlog right now.

File: app/models/account.rb
Comment (bradurani) [inline app/models/account.rb]: `Setting.reserved_usernames ||= []`
should be
`Setting.revserved_username || []`

**Review by bradurani — CHANGES_REQUESTED** [review]

**Comment by ashfurrow** [comment]
Thanks for the PR @ngerakines – looks good! Could you pull latest from master, review the changes, and address @bradurani 's comment above? Thanks again for your contribution!

File: app/models/account.rb
Comment (ngerakines) [inline app/models/account.rb]: Done!

**Comment by ngerakines** [comment]
The branch has been updated from current master and the change request applied.

**Comment by ashfurrow** [comment]
The code looks good, but there's a CI error. I've not seen it before, anyone have any ideas?

File: config/settings.yml
Comment (hikaru-light) [inline config/settings.yml]: LGTM:cake:

**Comment by hikaru-light** [comment]
I have another way because I have implemented reserving usernames in another service.
May I push my branch?

**Comment by Gargron** [comment]
The failure is because Setting.reserved_usernames is evaluated at class load time, which is before anything else, and in this case before the migrations finish running, so it runs a SELECT against an incomplete database.

**Pull request closed without merging**
